### PR TITLE
Curves animation

### PR DESCRIPTION
**Pull request by gspencergoog**
This converts the curve diagrams into animations so that users can see how the curves convert linear time into nonlinear output, with examples of how it can be used.

curve_elastic_in_out

(The "real" animation has a better frame rate: GitHub only supports GIFs).

### CODE REVIEW

**Comment by Hixie** [comment]
cc @InMatrix for advice

**Comment by InMatrix** [comment]
Here is an example you might want to take a look: https://easings.net/#easeInOutSine

I thought it looked clean and minimalist (in a good way).

**Comment by gspencergoog** [comment]
OK, I think this is ready for review.  I've run it past @InMatrix and incorporated his feedback.

I've also regenerated all the images, not just the animation curves, to make sure that they're all up to date with the current head of master.

**Comment by gspencergoog** [comment]
Note: this PR currently contains the changes in https://github.com/flutter/assets-for-api-docs/pull/36 as well: I'll merge those out as soon as the other PR lands.

File: diagrams/lib/curve.dart [hunk 1]
```diff
@@ -133,28 +203,237 @@ class CurveDescription extends CustomPainter {
...
+            ),
+          ),
+        ),
+        new Text(
+          name,
+          textAlign: TextAlign.center,
+          style: Theme.of(context).textTheme.body2.copyWith(
+                color: Colors.black,
+                fontSize: 12.0,
+              ),
```
Comment (Hixie) [inline diagrams/lib/curve.dart]: trivial nit: weird indentation

**Comment by Hixie** [comment]
LGTM

**Comment by Hixie** [comment]
Looks like Curves.elasticIn misses the last frame.
Also, there's something fishy about the line in ease_out, like it's loosing yarn or something.

**Pull request merged**